doc/user/content/sql/recursive-ctes.md

```diff
@@ -0,0 +1,297 @@
+
+## Details
+
+Within a recursive CTEs block, any `cte_ident` alias can be referenced in all `recursive_cte_binding` definitions that live under the same block, as well as in the final `select_stmt` for that block.
+
+Semantically, a recursive CTE block will initialize all of its bindings to the empty collection and sequentially update them in a loop until either:
+
+1. a fixpoint is reached (i.e., if the collections stop changing across iterations), or
+1. some early exit criteria is satisfied.
+
+{{< warning >}}
+In the absence of recursive CTEs, every `SELECT` query is guaranteed to compute its result or fail with an error within a finite amount of time.
+However, introducing recursive CTEs complicates the situation as follows:
+
+1. The query might not converge (and may never terminate).
+   Most likely, this will manifest in a query that never yields results and eats up a lot of your cluster resources. See [an example](#non-terminating-queries) below.
+2. A small update to a few (or even one) data points in your input might cascade in big updates in your recursive computation.
+   This most likely will manifest in spikes of the cluster resources allocated to your recursive dataflows.
+   See [an example](#queries-with-update-locality) below.
+{{</ warning >}}
+
+## Examples
+
+Let's consider a very simple schema consisting of `users` that belong to a hierarchy of geographical `areas` and exchange `transfers` between each other.
+
+
+### Example schema
+
+```sql
+-- A hierarchy of geographical locations with various levels of granularity.
+CREATE TABLE areas(id int not null, parent int, name text);
+-- A collection of users.
+CREATE TABLE users(id char(1) not null, area_id int not null, name text);
+-- A collection of transfers between these users.
+CREATE TABLE transfers(src_id char(1), tgt_id char(1), amount numeric, ts timestamp);
+```
+
+### Example data
+
+```sql
+DELETE FROM areas;
+DELETE FROM users;
+DELETE FROM transfers;
+
+INSERT INTO areas VALUES
+  (1, 2    , 'Brooklyn'),
+  (2, 3    , 'New York'),
+  (3, 7    , 'United States of America'),
+  (4, 5    , 'Kreuzberg'),
+  (5, 6    , 'Berlin'),
+  (6, 7    , 'Germany'),
+  (7, null , 'Earth');
+INSERT INTO users VALUES
+  ('A', 1, 'Alice'),
+  ('B', 4, 'Bob'),
+  ('C', 2, 'Carol'),
+  ('D', 5, 'Dan');
+INSERT INTO transfers VALUES
+  ('B', 'C', 20.0 , now() + '10 seconds'),
+  ('A', 'D', 15.0 , now() + '20 seconds'),
+  ('C', 'D', 25.0 , now() + '30 seconds'),
+  ('A', 'B', 10.0 , now() + '40 seconds'),
+  ('C', 'A', 35.0 , now() + '50 seconds');
+```
+
+### Transitive closure
+
+The following view will compute `reachable` as the transitive closure of a graph where:
+* each `user` is a graph vertex, and
+* a graph edge between users `x` and `y` exists and ONLY one transfer was sent recently from `a` to `b` (using the rather small `10 seconds` period here for the sake of illustration):
```

Comment:
There is "`x` and `y`", but "`a` to `b`".

Also, the change from "if and only if at least" to "and ONLY" completely changed the meaning. You could change it to simply "if at least", which has the intended meaning but avoids confusing non-maths people with the "if and only if" phrase.

Reply:
Changed to 

> exists only if a recent transfer from `x` to `y` has been made